musica-fortran/src/micm/micm_core.F90

```diff
@@ -1,28 +1,39 @@
-module musica_micm
+module micm_core
+    ! Top-level MICM interface
+
     use iso_c_binding
+    
     implicit none
 
     private
     public :: micm_t
 
     include "micm_c_def.F90"
 
-    type micm_t
+    type :: micm_t
+    ! The micm_t class defines the API
         private
         type(c_ptr) :: ptr
     contains
+        ! Creates solver from configure file
         procedure :: create_solver => micm_create_solver
```

Comment:
is this function needed? or can this all be handled in the constructor (`create_micm`)?